forc/src/utils/helpers.rs

```diff
@@ -52,10 +53,71 @@ pub fn read_manifest(manifest_dir: &Path) -> Result<Manifest, String> {
             ))
         }
     };
-    match toml::from_str(&manifest) {
+
+    let manifest = match toml::from_str(&manifest) {
         Ok(o) => Ok(o),
         Err(e) => Err(format!("Error parsing manifest: {}.", e)),
+    }?;
+
+    validate_manifest(manifest)
+}
+
+// Using (https://github.com/rust-lang/cargo/blob/489b66f2e458404a10d7824194d3ded94bc1f4e4/src/cargo/util/toml/mod.rs +
+// https://github.com/rust-lang/cargo/blob/489b66f2e458404a10d7824194d3ded94bc1f4e4/src/cargo/ops/cargo_new.rs) for reference
+
+fn validate_name(name: &str, use_case: &str) -> Result<(), String> {
+    // if true returns formatted error
+    restricted_names::contains_invalid_char(name, use_case)?;
+
+    if restricted_names::is_keyword(name) {
+        return Err(format!(
+            "the name `{name}` cannot be used as a package name, it is a Sway keyword"
+        ));
+    }
+    if restricted_names::is_conflicting_artifact_name(name) {
+        return Err(format!(
+            "the name `{name}` cannot be used as a package name, \
+            it conflicts with Forc's build directory names"
+        ));
+    }
+    if name == "test" {
+        return Err("the name `test` cannot be used as a package name, \
+            it conflicts with Sway's built-in test library"
+            .into());
```

Comment:
small nit, the other errors are using `format!` and this one is calling `.into()`. Would be nice to keep consistency

Reply:
Ah, I had it as `.into()` since there wasn't any variables to insert into the string. Should I still make the change?

Reply:
Sorry, you're right, I didn't see the fancy new {variable} syntax that rust has now. Na this is all good how you had it.